spec/controllers/users_controller_spec.rb

```diff
@@ -31,73 +31,117 @@
   end
 
   describe "POST create" do
-    before do
-      allow(Stripe::Charge).to receive(:create)
-    end
-
-    context "with valid credentials" do
+    context "with valid personal information" do
       after { ActionMailer::Base.deliveries.clear }
 
-      it "calls Stripe to charge user with subscription fees" do
-        expect(Stripe::Charge).to receive(:create)
-        post :create, user: Fabricate.attributes_for(:user)
+      before do
+        expect(StripeWrapper::Charge).to receive(:create).and_return(charge)
       end
 
-      it "creates new user" do
-        post :create, user: Fabricate.attributes_for(:user)
-        expect(User.count).to eq(1)
-      end
+      context "and valid credit card" do
+        let(:charge) do
+          double(:charge,
+                 successful?: true)
+        end
 
-      it "redirects to the sign-in page" do
-        post :create, user: Fabricate.attributes_for(:user)
-        expect(response).to redirect_to(sign_in_path)
+        it "creates new user" do
+          post :create, user: Fabricate.attributes_for(:user)
+          expect(User.count).to eq(1)
+        end
+
+        it "redirects to the sign-in page" do
+          post :create, user: Fabricate.attributes_for(:user)
+          expect(response).to redirect_to(sign_in_path)
+        end
+
+        context "with invitation" do
```

Comment:
Thanks @falonofthetower for the review!
I've incorporated `let`s and `before`s where it made sense... i had to de-nest the `context 'with invitation'` for their effective use.
Totally agree with your readability point. I avoid `let`s and `before`s as I start out writing test because i am worried DRYing tests this way would affect readability adversely, especially when it detaches the test setup (`let`s) and action (`before`s) too much from the assertion.
Let me know if that has happened with the refactoring.